src/meili/routing.cc

```diff
@@ -529,70 +526,72 @@ find_shortest_path(baldr::GraphReader& reader,
         break;
       }
 
+      // If this isnt the origin label then we dont need to queue any edges from it
+      // we only needed to check if it was a yet to be found destination above
+      if (destination_idx != origin_idx)
+        continue;
+
       // Expand origin: add segments from origin to destinations ahead
       // at the same edge as well as at the opposite edge to the queue
-      if (destination_idx == origin_idx) {
```

Comment:
the diff below is mostly just an indentation change from adding that continue statement above